torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -881,7 +881,10 @@ def _find_onnx_data_type(
     if isinstance(torch_input, (int, float, bool, str, complex)):
         return fx_type_utils.from_torch_dtype_to_onnx_dtype_str(type(torch_input))
     if isinstance(torch_input, (list, tuple)) and torch_input:  # [Tensor, Tensor]
-        set_dtype = _find_onnx_data_type(torch_input[0])
+        the_first_non_none_item = next(
+            (item for item in torch_input if item is not None), None
+        )
+        set_dtype = _find_onnx_data_type(the_first_non_none_item)
```

Comment:
What if it is all None? Should we match _any_ type in that case?

Reply:
It's going to hit the None if branch down below, which matches all types as the comments suggesting.